test/components/DayPickerSingleDateController_spec.jsx

```diff
@@ -974,6 +974,19 @@ describe('DayPickerSingleDateController', () => {
       expect(firstFocusableDay.isSame(date, 'day')).to.equal(true);
     });
 
+    it('time is a noon', () => {
+      sinon.stub(DayPickerSingleDateController.prototype, 'isBlocked').returns(false);
+      const wrapper = shallow((
+        <DayPickerSingleDateController
+          date={null}
+          onDateChange={sinon.stub()}
+          onFocusChange={sinon.stub()}
+        />
+      ));
+      const firstFocusableDay = wrapper.instance().getFirstFocusableDay(today);
+      expect(firstFocusableDay.hours()).to.equal(12);
+    });
```

Comment:
this seems like it's testing the implementation, rather than the bug reported in #2140.

Reply:
@ljharb, not sure I follow what you mean.
From bug:
> expected behavior: returns consistent time, either always midnight local time or always noon local time
Test ensures that time is a noon. Bug closed.

Reply:
right, but the bug is "it's called twice and the time isn't the same" - so shouldn't the test call it twice and guarantee the time is the same?

(it's fine to *also* test it's at noon, because there's other reasons it should be noon)

Reply:
"called twice" resolved in another commit, I sent you another PR with cache solution to resolve it. This PR fixes time issue only.

Reply:
The issue was that *when* it's called multiple times, even across renders, the *time* is different. I don't think the caching would address that.

Reply:
@ljharb As I said, no caching involved in my solution in current PR. I reverted it. The only changes - to set up hours to 12:00.